spark/src/test/scala/org/apache/spark/sql/delta/rowid/RowTrackingUpdateSuite.scala

```diff
@@ -0,0 +1,291 @@
+      None // No condition, 100% of files match
+    )
+  } {
+    test(s"Preserves row IDs, whereClause = $whereClause, isPartitioned = $isPartitioned") {
+      withRowIdTestTable(isPartitioned = isPartitioned) {
+        checkAndExecuteUpdate(tableName = targetTableName, condition = whereClause)
+      }
+    }
+  }
+
+  for (isPartitioned <- BOOLEAN_DOMAIN)
+  test(s"Preserves row IDs across multiple updates, isPartitioned = $isPartitioned") {
+    withRowIdTestTable(isPartitioned = false) {
+      checkAndExecuteUpdate(targetTableName, condition = Some("id % 20 = 0"))
+
+      checkAndExecuteUpdate(targetTableName, condition = Some("id % 10 = 0"), newVersion = 2L)
+    }
+  }
+
+  test("Preserves row IDs in update on partition column, whole file update") {
+    withRowIdTestTable(isPartitioned = true) {
+      checkAndExecuteUpdate(tableName = targetTableName, condition = Some("partition = 0"))
+    }
+  }
+
+
+  test(s"Preserves row IDs on unpartitioned table with optimized writes") {
+    withRowIdTestTable(isPartitioned = false) {
+      val whereClause = Some(s"id = 0 OR id = $numRowsTarget - 1")
+      withSQLConf(
+        DeltaSQLConf.DELTA_OPTIMIZE_WRITE_ENABLED.key -> "true") {
+        checkAndExecuteUpdate(targetTableName, condition = whereClause)
+      }
+
+      val (_, snapshot) = DeltaLog.forTableWithSnapshot(spark, TableIdentifier(targetTableName))
+
+      val expectedNumFiles = if (dvsEnabled) numFiles + 1 else numFiles - 1
+      assert(snapshot.allFiles.count() === expectedNumFiles)
+    }
+  }
+
+  test("Row tracking marked as not preserved when row tracking disabled") {
+    withRowTrackingEnabled(enabled = false) {
+      withRowIdTestTable(isPartitioned = false) {
+        val log = DeltaLog.forTable(spark, TableIdentifier(targetTableName))
+        assert(
+          !rowTrackingMarkedAsPreservedForCommit(log)(executeUpdate(
+            targetTableName, where = None, newVersion = -1L)))
+      }
+    }
+  }
+
+  test("Preserving Row Tracking - Subqueries are not supported in UPDATE") {
+    withRowTrackingEnabled(enabled = true) {
+      withRowIdTestTable(isPartitioned = false) {
+        val ex = intercept[AnalysisException] {
+          checkAndExecuteUpdate(
+            tableName = targetTableName,
+            condition = Some(
+              s"""id in (SELECT id FROM $targetTableName s
+              WHERE s.id = 0 OR s.id = $numRowsPerFile)"""))
+        }.getMessage
+        assert(ex.contains("Subqueries are not supported in the UPDATE"))
+      }
+    }
+  }
+}
+
+trait RowTrackingUpdateDVTests extends RowTrackingUpdateSuiteBase
+  with DeletionVectorsTestUtils {
+
+  override protected def dvsEnabled: Boolean = true
+
+}
+
+trait RowTrackingCDFTests extends RowTrackingUpdateSuiteBase {
+
+  override protected def sparkConf: SparkConf = {
+    super.sparkConf.set(DeltaConfigs.CHANGE_DATA_FEED.defaultTablePropertyKey, "true")
+  }
+}
+
+class RowTrackingUpdateSuite extends RowTrackingUpdateCommonTests
```

Comment:
That's a lot of test suites, I wonder if we shouldn't  trim it a bit and focus on suite that provide high coverage